xds/src/main/java/io/grpc/xds/internal/security/trust/XdsX509TrustManager.java

```diff
@@ -245,19 +258,37 @@ public void checkServerTrusted(X509Certificate[] chain, String authType, SSLEngi
       sslParams.setEndpointIdentificationAlgorithm("");
       sslEngine.setSSLParameters(sslParams);
     }
-    delegate.checkServerTrusted(chain, authType, sslEngine);
+    chooseDelegate(chain).checkServerTrusted(chain, authType, sslEngine);
     verifySubjectAltNameInChain(chain);
   }
 
   @Override
   public void checkServerTrusted(X509Certificate[] chain, String authType)
       throws CertificateException {
-    delegate.checkServerTrusted(chain, authType);
+    chooseDelegate(chain).checkServerTrusted(chain, authType);
     verifySubjectAltNameInChain(chain);
   }
 
+  private X509ExtendedTrustManager chooseDelegate(X509Certificate[] chain)
+      throws CertificateException {
+    if (delegates != null) {
+      Optional<SpiffeUtil.SpiffeId> spiffeId = SpiffeUtil.extractSpiffeId(chain);
+      if (!spiffeId.isPresent()) {
+        throw new CertificateException("Can't extract valid SPIFFE ID from the chain");
+      }
+      String trustDomain = spiffeId.get().getTrustDomain();
+      if (!delegates.containsKey(trustDomain)) {
+        throw new CertificateException(
+            "Spiffe Trust Bundle doesn't contain trust domain from the chain");
+      }
+      return delegates.get(trustDomain);
+    } else {
+      return delegate;
+    }
+  }
+
   @Override
   public X509Certificate[] getAcceptedIssuers() {
-    return delegate.getAcceptedIssuers();
+    return delegates == null ? delegate.getAcceptedIssuers() : null;
```

Comment:
When `delegates != null`, shouldn't we be returning the union of all accepted issuers from all of the delegates in the map?